documentation/document-api-guide.html

```diff
@@ -50,14 +48,75 @@ <h2 id="documents">Documents</h2>
+DocumentAccess</a> javadoc. Sample app:
 <pre>
-DocumentAccess access = DocumentAccess.createDefault();
-DocumentType type = access.getDocumentTypeManager().getDocumentType("music");
-DocumentId id = new DocumentId("id:music:music::0");
-Document document = new Document(type, id);
-//Do great stuff, then remember to shutdown the Access instance  
-access.shutdown();
+&lt;dependencies&gt;
+    &lt;dependency&gt;
+        &lt;groupId&gt;com.yahoo.vespa&lt;/groupId&gt;
+        &lt;artifactId&gt;documentapi&lt;/artifactId&gt;
+        &lt;version&gt;7.168.4&lt;/version&gt; &lt;!-- Find latest version at <a href="https://search.maven.org/search?q=g:com.yahoo.vespa%20a:documentapi">search.maven.org/search?q=g:com.yahoo.vespa%20a:documentapi</a> --&gt;
+    &lt;/dependency&gt;
+    &lt;dependency&gt; &lt;!-- ToDo: remove this once ImmutableList is removed --&gt;
+        &lt;groupId&gt;>com.google.guava&lt;/groupId&gt;
+        &lt;artifactId&gt;guava&lt;/artifactId&gt;
+        &lt;version&gt;28.2-jre&lt;/version&gt;
+    &lt;/dependency&gt;
+&lt;dependencies&gt;
+</pre>
+<pre>
+import com.yahoo.document.DataType;
+import com.yahoo.document.Document;
+import com.yahoo.document.DocumentId;
+import com.yahoo.document.DocumentPut;
+import com.yahoo.document.DocumentType;
+import com.yahoo.document.DocumentUpdate;
+import com.yahoo.document.datatypes.StringFieldValue;
+import com.yahoo.document.datatypes.WeightedSet;
+import com.yahoo.document.update.FieldUpdate;
+import com.yahoo.documentapi.DocumentAccess;
+import com.yahoo.documentapi.SyncParameters;
+import com.yahoo.documentapi.SyncSession;
+
+public class DocClient {
+
+    public static void main(String[] args) {
+
+        DocumentAccess access = DocumentAccess.createDefault();
+        DocumentType type = access.getDocumentTypeManager().getDocumentType("music");
+        DocumentId id = new DocumentId("id:namespace:music::0");
+        Document doc = new Document(type, id);
+        SyncSession session = access.createSyncSession(new SyncParameters.Builder().build());
+
+        // Put document with a1,1
+        WeightedSet&lt;StringFieldValue&gt; wset = new WeightedSet&lt;&gt;(DataType.getWeightedSet(DataType.STRING));
+        wset.put(new StringFieldValue("a1"), 1);
+        doc.setFieldValue("aWeightedset", wset);
+        DocumentPut put = new DocumentPut(doc);
+        System.out.println(doc.toJson());
+        session.put(put);
+
+        // Update document with a1,10 and a2,20
+        DocumentUpdate upd1 = new DocumentUpdate(type, id);
+        WeightedSet&lt;StringFieldValue&gt; wset1 = new WeightedSet&lt;&gt;(DataType.getWeightedSet(DataType.STRING));
+        wset1.put(new StringFieldValue("a1"), 10);
+        wset1.put(new StringFieldValue("a2"), 20);
+        upd1.addFieldUpdate(FieldUpdate.createAddAll(type.getField("aWeightedset"), wset1));
+        System.out.println(upd1.toString());
+        session.update(upd1);
+
```

Comment:
Could be nice to have a final step here where we Get the document back out again and print it, showing that the operations have taken effect.

Reply:
great idea, will add in next iteration !